components/script/dom/webgl2renderingcontext.rs

```diff
@@ -2914,6 +2916,100 @@ impl WebGL2RenderingContextMethods for WebGL2RenderingContext {
             .TexImage2D_(target, level, internal_format, format, data_type, source)
     }
 
+    /// https://www.khronos.org/registry/webgl/specs/latest/2.0/#3.7.6
+    #[allow(unsafe_code)]
+    fn TexImage2D__(
+        &self,
+        target: u32,
+        level: i32,
+        internalformat: i32,
+        width: i32,
+        height: i32,
+        border: i32,
+        format: u32,
+        type_: u32,
+        src_data: CustomAutoRooterGuard<ArrayBufferView>,
+        src_offset: u32,
+    ) -> Fallible<()> {
+        if self.bound_pixel_unpack_buffer.get().is_some() {
+            return Ok(self.base.webgl_error(InvalidOperation));
+        }
+
+        if type_ == constants::FLOAT_32_UNSIGNED_INT_24_8_REV && src_data.len() != 0 {
```

Comment:
We shouldn't check src_data.len() at all - this method only takes a non-null argument, so we should always report an error if this type is used in this method.

Reply:
Thank you! I've updated the pull request with this change.